### PR TITLE
WT-11445 Create additional stats for the background compaction server

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

**Comment by Sean04** [comment]
I think we should add a few more such as:
- `bytes_rewritten`
- `bytes_recovered`
- `background_compact.bytes_rewritten_ema`

**Comment by etienneptl** [comment]
> I think we should add a few more such as:
> 
> * `bytes_rewritten`
> * `bytes_recovered`
> * `background_compact.bytes_rewritten_ema`

Thanks for the suggestions! See below:

`bytes_rewritten` -> 2598e5acd885cc6002c35ce15982ac5be1993a8a
`bytes_recovered` -> 0e0bcfe713a875ef644c1f91ff732698b167fdda
`background_compact.bytes_rewritten_ema` -> 4899917d221681337dd90af89d6f8301c41f0e55

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/btree/bt_compact.c [hunk 1]
```diff
@@ -334,9 +334,6 @@ __wt_compact(WT_SESSION_IMPL *session)
         WT_STAT_CONN_INCR(session, session_table_compact_skipped);
         WT_STAT_DATA_INCR(session, btree_compact_skipped);
 
-        if (session == S2C(session)->background_compact.session)
-            WT_STAT_CONN_INCR(session, background_compact_skipped);
```
Comment (etienneptl) [inline src/btree/bt_compact.c:338]: This was removed as `session_table_compact_skipped` just above already covers this. We don't necessarily need to distinguish between fg and bg compaction.

**Review by Sean04 — APPROVED** [review]
LGTM

**Comment by etienneptl** [comment]
evergreen merge

**Pull request merged**